proposals/NNNN-psv0-entry-name.md

```diff
@@ -0,0 +1,213 @@
+object API usage scenarios.
+
+With this solution, when a new compiler and validator are used to compile an
+application's shaders, these shaders will still be compatible with older
+pipeline API and runtime versions while including the additional information
+that makes them convenient to use in the new state object API.
+
+## Detailed design
+
+### PSV0 data layout
+
+The data layout for `PSV0` is described with pseudo-code
+[here](https://github.com/microsoft/DirectXShaderCompiler/blob/e496555aae0c9efdc67430994055a8077a8b86cc/include/dxc/DxilContainer/DxilPipelineStateValidation.h#L821).
+Like all other DxilContainer parts, values in `PSV0` are little-endian.
+
+### Versioning PSV0 for the new data
+
+The `PSV0` part has a versioning convention that allows backward and forward
+compatibility, and is tied to validator version, as opposed to shader model.
+Data added to the format will be safely ignored by an older runtime.
+
+This works as follows:
+
+- A version number determines which version of the top-level `PSVRuntimeInfoN`
+  structure will be used, based on the validator version.  This version also
+  determines what additional information will be included in the part.
+- Adding new data is a matter of deriving from the last version of the
+  `PSVRuntimeInfoN` structure, incrementing the number at the end, and adding
+  the needed fields to the new structure.
+- The structure size is the first element of the `PSV0` part, indicating to a
+  reader the available versions in the serialized data.  If that size is larger
+  than the newest structure a reader knows about, it must ignore additional
+  data after the end of the newest version of `PSVRuntimeInfoN` that it knows
+  about.
+- Additional data (such as record tables) needed for newer versions of the
+  structure will appear after this additional data for previous versions of the
+  structure.
+
+Hooking up a new `PSVRuntimeInfoN` version requires:
+- updating the `MAX_PSV_VERSION` and the `RuntimeInfoSize()` function
+- adding a pointer for the new structure to `DxilPipelineStateValidation`
+- updating `ReadOrWrite` method to use `AssignDerived` for the new pointer
+- updating `DxilContainerAssembler` to use the new version based on the next
+  validator version, so containers produced will work with existing validators.
+
+Making the new data available requires adding any needed accessors to
+`DxilPipelineStateValidation` and writing the new data from
+`DxilContainerAssembler`, when a new version is selected.
+
+In this case this consists of:
+- Adding one accessor that returns a null-terminated utf-8 string pointer, by
+  looking it up in the string table at the offset provided by the new field in
+  the new version of the `PSVRuntimeInfoN` structure.
+- In `DxilContainerAssembler`, when writing the new version of `PSV0`, add the
+  entry function name to the string table, and write the offset a new
+  `EntryFunctionName` field in the new `PSVRuntimeInfoN` structure.
+
```

Comment:
This "Detailed Design" section does not actually describe the concrete changes you're proposing. The design should provide enough information that I can implement this change in LLVM without needing to look at DXC's codebase to reverse engineer the new structures.

Reply:
> The design should provide enough information that I can implement this change in LLVM without needing to look at DXC's codebase to reverse engineer the new structures.

In order to implement "this change" (one added field), you must have implemented the rest of PSV0.  "concrete changes" require a concrete context to reference.  However, I have nothing concrete to refer to other than the DXC codebase, since that's where PSV0 is currently defined.  If I cannot reference that, I must specify all of PSV0 independently from DXC.  Yet it was made clear that this was not required for this proposal.

I linked to the comment that describes the data layout for PSV0 above.  I described the elements this spec interacts with, how the versioning works, and the concrete changes necessary to add this field to a derived structure, as well as other implementation details for DXC.  I linked to a PR implementing this proposal.  Yes, these refer to the DXC codebase.  That's because the DXC codebase is the only definition I can reference.

For the purpose of evaluating this proposal, what I've included should be more than sufficient.  If we want an independent specification for PSV0, we can spend time writing that.  I do not think that is required to evaluate this proposal.

By the way: it's not called "reverse engineering" when you have the full source code and comments describing layouts and intentions in a working project you can build.

I can think of things I could do to localize more of the specifics referred to here, but none of that will satisfy the "enough information" requirement above unless I end up fully specifying PSV0 here.

Can you be more specific about what you think this proposal should include and/or exclude based on what's here so far?